app/controllers/orders_controller.rb

```diff
@@ -60,6 +70,9 @@ def purchase
 
   # only partial params, update if more attributes are required.
   def order_params
-    params.require(:order).permit(:shopper_name, :shopper_email, :shopper_address, :cc_exp)
+    if params[:order][:cc_all]
+      params[:order][:cc_four] = params[:order][:cc_all][-4..-1] # front-end validation requires min 4 char input.
+    end
```

Comment:
Did this used to be in the model? For my own learning, I'd be curious to hear about why this is here, if you have the bandwidth/desire to tell me! :-)